### PR TITLE
[SL-UP] 917ncp init failure fix

### PR DESCRIPTION
**Pull request by arun-silabs**
This PR fixes the init failure in 917NCP boards.
After integrating the matter_sdk code with the sl_net APIs, the init is failing because LWIP init is called twice for 917NCP. 
To resolve this, we have removed the LWIP init from PlatformManager and modified the init sequence to avoid errors while booting. But, with the recent SPI related changes, we are facing SPI_BUSY error. So, reverted the recent changes made in efx32_ncp_host.c file. With these changes, the init for 917NCP is passing.

### CODE REVIEW

**Comment by mkardous-silabs** [comment]
Why does the efx32_ncp_host update cause errors? Did we root cause the the seen errors?

**Comment by arun-silabs** [comment]
> Why does the efx32_ncp_host update cause errors? Did we root cause the the seen errors?

The error is seen during SPI transfer, sl_si91x_host_spi_transfer() - This API is throwing SPI_BUSY error. I've tried to debug, but couldn't root cause it. 
The SPI issue with the latest efx32_ncp_host.c file from wifi_sdk is coming up after integrating with sl_net layer. Need to debug the exact rootcause.

**Comment by bhmanda-silabs** [comment]
> > > Why does the efx32_ncp_host update cause errors? Did we root cause the the seen errors?
> > 
> > 
> > The error is seen during SPI transfer, sl_si91x_host_spi_transfer() - This API is throwing SPI_BUSY error. I've tried to debug, but couldn't root cause it. The SPI issue with the latest efx32_ncp_host.c file from wifi_sdk is coming up after integrating with sl_net layer. Need to debug the exact rootcause.
> 
> Let me know when we have a better idea with what is happening with the latest efx32 ncp host file and the sl_net APIs. It might be worth while to open a Wiseconnect issue for this to get them involved.

Wiseconnect ticket - https://jira.silabs.com/browse/WIFISDK-1109

File: src/platform/silabs/wifi/rs911x/platform/efx32_ncp_host.c [hunk 1]
```diff
@@ -42,18 +40,20 @@
...
 
-#define MAX_DATA_PACKET_SIZE 1800
 #define LDMA_MAX_TRANSFER_LENGTH 4096
 #define LDMA_DESCRIPTOR_ARRAY_LENGTH (LDMA_MAX_TRANSFER_LENGTH / 2048)
 
 // use SPI handle for EXP header (configured in project settings)
 extern SPIDRV_Handle_t sl_spidrv_exp_handle;
 #define SPI_HANDLE sl_spidrv_exp_handle
-static uint8_t dummy_buffer[MAX_DATA_PACKET_SIZE] = { 0 };
+static uint8_t dummy_buffer[1800] = { 0 };
```
Comment (mkardous-silabs) [inline src/platform/silabs/wifi/rs911x/platform/efx32_ncp_host.c:56]:
```suggestion
static uint8_t dummy_buffer[MAX_DATA_PACKET_SIZE] = { 0 };

```

File: src/platform/silabs/wifi/rs911x/platform/efx32_ncp_host.c [hunk 2]
```diff
@@ -42,18 +40,20 @@
...
 #endif // SL_BOARD_NAME
 
+#include "sl_si91x_ncp_utility.h"
+#include <platform/silabs/wifi/wf200/platform/spi_multiplex.h>
+
 #include "sl_spidrv_exp_config.h"
 #include "sl_spidrv_instances.h"
 #include "spidrv.h"
 
-#define MAX_DATA_PACKET_SIZE 1800
```
Comment (mkardous-silabs) [inline src/platform/silabs/wifi/rs911x/platform/efx32_ncp_host.c:49]: can we keep this define for clarify?

File: src/platform/silabs/wifi/rs911x/platform/efx32_ncp_host.c [hunk 3]
```diff
@@ -42,18 +40,20 @@
 #include "sl_board_control.h"
 #endif // SL_BOARD_NAME
 
+#include "sl_si91x_ncp_utility.h"
+#include <platform/silabs/wifi/wf200/platform/spi_multiplex.h>
```
Comment (mkardous-silabs) [inline src/platform/silabs/wifi/rs911x/platform/efx32_ncp_host.c:44]: These don't need to be seperate - can we keep them with the rest?

**Review by mkardous-silabs — APPROVED** [review]

File: src/platform/silabs/wifi/rs911x/platform/efx32_ncp_host.c (hunk 1 above)
Comment (bhmanda-silabs) [reply src/platform/silabs/wifi/rs911x/platform/efx32_ncp_host.c]: done

File: src/platform/silabs/wifi/rs911x/platform/efx32_ncp_host.c (hunk 2 above)
Comment (bhmanda-silabs) [reply src/platform/silabs/wifi/rs911x/platform/efx32_ncp_host.c:49]: done

File: src/platform/silabs/wifi/rs911x/platform/efx32_ncp_host.c (hunk 3 above)
Comment (bhmanda-silabs) [reply src/platform/silabs/wifi/rs911x/platform/efx32_ncp_host.c]: Updated as per comment

**Review by senthilku — APPROVED** [review]

**Review by bhmanda-silabs — APPROVED** [review]

**Pull request merged**